analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -0,0 +1,533 @@
+        // Currently we do not support usage of two different pub-cache directories.
+        // Therefore you need to delete the .pub-cache directory in your flutter root folder
+        // to force flutter to use the default one
+        private val pubCacheRoot = getUserHomeDirectory().resolve(".pub-cache/")
+
+        fun findFile(packageInfo: JsonNode, fileName: String): File? {
+            val artifactRootDir = findProjectRoot(packageInfo);
+
+            // try to locate file directly
+            val file = File(
+                artifactRootDir,
+                fileName
+            );
+            if(file.isFile) return file;
+
+            // search directory for file
+            return artifactRootDir.walkTopDown().find {
+                it.isFile && it.name == fileName
+            }
+        }
+
+        fun findProjectRoot(packageInfo: JsonNode): File {
+            var path : String;
+
+            val packageVersion = packageInfo["version"].textValueOrEmpty();
+            val type = packageInfo["source"].textValueOrEmpty();
+            val description = packageInfo["description"];
+            val packageName = description["name"].textValueOrEmpty();
+            val url = description["url"].textValueOrEmpty();
+            val resolvedRef = packageInfo["resolved-ref"].textValueOrEmpty();
+
+            if(type == "hosted" && url.isNotEmpty()){
+                // packages with source set to "hosted" and "url" key in description set to "https://pub.dartlang.org",
+                // path should be resolved to "hosted/pub.dartlang.org/packageName-packageVersion"
+                path = "hosted/" + url.replace("https://","") + "/" + packageName + "-" + packageVersion;
+            } else if(type == "git" && resolvedRef.isNotEmpty()){
+                // packages with source set to "git" and a "resolved-ref" key in description set to a gitHash,
+                // should be resolved to "git/packageName-gitHash"
+                path = "git/" + packageName + "-" + resolvedRef;
+            } else {
+                // not supported type
+                path = "/";
+
+                log.error { "Could not find projectRoot of '$packageName'" }
+            }
+
+            val artifactRootDir = File(
+                pubCacheRoot,
+                path
+            )
+
+            return artifactRootDir;
+        }
+    }
+
+    private val reader = PubCacheReader();
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        val workingDir = definitionFile.parentFile
+
+        val manifest = yamlMapper.readTree(definitionFile)
+        val hasDependencies = manifest.fields().asSequence().any { (key, value) ->
+            key.startsWith("dependencies") && value.count() > 0
+        }
+
+        val (packages, scopes) = if (hasDependencies) {
+            installDependencies(workingDir)
+
+            log.info { "Reading $PUB_LOCK_FILE file in $workingDir " }
```

Comment:
End the message with a dot and remove the trailing slashes in this log message and the three below.